doc/system-telemetry/reboot-cause.md

```diff
@@ -0,0 +1,150 @@
+# Reboot-cause information via telemetry agent
+
+## Revision
+
+| Rev | Date     | Author      | Change Description |
+|:---:|:--------:|:-----------:|--------------------|
+| 0.1 | 09/02/20 | Sujin Kang  | Initial version    |
+
+## Scope
+-Enable sonic streaming telemetry agent to send Reboot-cause information
+
+### Enable sonic streaming telemetry agent to send Reboot-cause information
+
+#### Part 1
+During the boot, the `determine-reboot-cause` service ( previously `process-reboot-cause`) determines the last reboot-cause based on the hardware reboot-cause
+and the software reboot-cause information and `determine-reboot-cause` service will save the formatted last previous
+reboot cause information to "/host/reboot-cause/previous-reboot-cause/" with adding timestamp at the end of file name.
+`determine-reboot-cause` also will also create a symbolic link of the last reboot cause file to "/host/reboot-cause/previous-reboot-cause.txt"
```

Comment:
Do we need previous-reboot-cause.txt any longer, when we could just read the file with the most recent timestamp?

Reply:
To make the `show reboot-cause`  simple, otherwise, we need to read file list in the directory and sort them before parsing the files for displaying the cli output. Before it was just one step, just "cat /host/reboot-cause/previous-reboot-cause.txt" and now it's two steps - parsing and composing as output format. If we just using the file reading from the most recent timestamp, it will be more than 4 steps - read filelist, sort them in timeorder, pase the file and formatting as output for each cli call.

Reply:
Understood. Then I'm OK with symlinking the latest file. However, this file, once symlinked, will no longer be a plain text file, so we should change the suffix.

Also, I suggest changing the naming of the dirs/files as such:

`/host/reboot-cause/previous-reboot-cause/` --> `/host/reboot-cause/history/`
`/host/reboot-cause/previous-reboot-cause.txt` --> `/host/reboot-cause/previous-reboot-cause.json` (or simply `/host/reboot-cause/previous.json`